widget/list.go

```diff
@@ -207,6 +207,13 @@ func (l *listRenderer) Layout(size fyne.Size) {
 		return
 	}
 	min := int(fyne.Min(float32(length), float32(l.visibleItemCount)))
+	if length < len(l.children) {
+		l.firstItemIndex = 0
```

Comment:
I think this code runs when the dataset has shrunk right?
I don't think that this means we should scroll to top.
The only scrolling that should be required is if our visible viewport is beyond the space now required for the list.

Reply:
Fair enough. I will remove the scroll to top.

Reply:
done